messenger.rst

```diff
@@ -501,112 +467,84 @@ you can disable them like this:
-                        default_middleware: false
+                transports:
+                    async_priority_high:
+                        dsn: '%env(MESSENGER_TRANSPORT_DSN)%'
+                        options:
+                            # queue_name is specific to the doctrine transport
+                            # try "exchange" for amqp or "group1" for redis
+                            queue_name: high
+                    async_priority_low:
+                        dsn: '%env(MESSENGER_TRANSPORT_DSN)%'
+                        options:
+                            queue_name: high
 
-    .. code-block:: xml
+                routing:
+                    'App\Message\SmsNotification':  async_priority_low
+                    'App\Message\NewUserWelcomeEmail':  async_priority_high
 
-        <!-- config/packages/messenger.xml -->
-        <?xml version="1.0" encoding="UTF-8" ?>
-        <container xmlns="http://symfony.com/schema/dic/services"
-            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-            xmlns:framework="http://symfony.com/schema/dic/symfony"
-            xsi:schemaLocation="http://symfony.com/schema/dic/services
-                https://symfony.com/schema/dic/services/services-1.0.xsd
-                http://symfony.com/schema/dic/symfony
-                https://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+You can then run individual workers for each transport or instruct one worker
+to handle messages in a priority order:
 
-            <framework:config>
-                <framework:messenger>
-                    <framework:bus name="messenger.bus.default" default-middleware="false"/>
-                </framework:messenger>
-            </framework:config>
-        </container>
+.. code-block:: terminal
 
-    .. code-block:: php
+    # consume async_lo
+    $ php bin/console messenger:consume async_priority_high async_priority_low
 
-        // config/packages/messenger.php
-        $container->loadFromExtension('framework', [
-            'messenger' => [
-                'buses' => [
-                    'messenger.bus.default' => [
-                        'default_middleware' => false,
-                    ],
-                ],
-            ],
-        ]);
+The worker will always first look for messages waiting on ``async_priority_high``. If
+there are none, *then* it will consume messages from ``async_priority_low``.
 
-Adding your own Middleware
-~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. _messenger-supervisor:
 
-As described in the component documentation, you can add your own middleware
-within the buses to add some extra capabilities like this:
+Supervisor Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. configuration-block::
+Supervisor is a great tool to guarantee that your worker process(es) is
+*always* running (even if it closes due to failure, hitting a message limit
+or thanks to ``messenger:stop-workers``). You can install it on Ubuntu, for
```

Comment:
We should mention Windows and macOS too, even adding some links to how to install Supervisor (or its alternatives, if they are compatible with Messenger too).

Reply:
Usually, we do not show installation instructions of third party tools. We only link to their official installation guide

Reply:
It's so simple to say `apt-get install supervisor` - I think it illustrates (without going into too much detail) how you install it on the most popular system. The realistic options are just to say "Install supervisor" and nothing else (because we don't want to link to or show every system's setup, obviously) or `sudo apt-get install supervisor`. The latter feels more practical to me

Reply:
Hmm, good point! I think the best way is linking to their docs and showing the apt install line (btw please suggest `apt` and not `apt-get`), that'll please everyone :)